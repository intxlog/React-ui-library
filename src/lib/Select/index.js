import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

//import css
import styles from './styles.module.scss'

const Select = ({
  id,
  children,
  disabled,
  defaultValue,
  onChange,
  onBlur,
  error
}) => {
  return (
    <div
      className={styles.wrapper} 
    >
      <select
        id={id}
        className={classNames({
          [styles.element]: true,
          [styles.disabled]: disabled,
          [styles.error]: error
        })}
        defaultValue={defaultValue}
        disabled={disabled}
        onChange={onChange}
        onBlur={onBlur}
      >
        {children}
      </select>
    </div>
  )
}

//rules for props being passed in
Select.propTypes = {
  id: PropTypes.string,
  children: PropTypes.array,
  disabled: PropTypes.bool,
  onChange: PropTypes.func,
  onBlur: PropTypes.func,
  error: PropTypes.bool
}

//maintain the name for documentation purposes
Select.displayName = 'Select'

export default Select