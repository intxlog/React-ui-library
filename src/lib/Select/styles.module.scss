// imports
@import './../../styles/helpers.scss';

.element {
  background: transparent;
  border: 0;
  outline: 0;
  width: 100%;
  color: var(--colorFormInputFontColor);
  font-size: $fontSize;
  line-height: 20px;
  padding: 6px 10px;
  user-select: none;
  height: 32px;
  -webkit-appearance: none;
}

.wrapper {
  @extend %element;
  position: relative;
  overflow: hidden;
  background-color: white;

  &, > * {
    cursor: pointer;
  }

  &:after {
    content: '';
    width: 0;
    position: absolute;
    right: 7px;
    top: 50%;
    margin-top: -3px;
    border-width: 6px 4px;
    border-style: solid;
    pointer-events: none;
    border-color: $colorBorder transparent transparent transparent;
  }

  &.disabled {
    background-color: rgb(235, 235, 228);
  }
}