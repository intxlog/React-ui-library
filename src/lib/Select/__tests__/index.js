import React from 'react'
import { shallow } from 'enzyme'
import Select from '..'

describe('Select', () => {
  let props = {}
  
  //create a shallow wrapper that can be reused
  let wrapper
  //initialize the components
  beforeEach(() => {
    wrapper = shallow(<Select {...props}/>)
  })

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
})