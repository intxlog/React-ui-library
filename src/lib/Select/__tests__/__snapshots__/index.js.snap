// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select renders correctly 1`] = `
<div
  className="wrapper"
>
  <select
    className="element"
  />
</div>
`;
